src/main/java/com/softserve/teamproject/entity/Student.java

```diff
@@ -111,7 +111,7 @@ public boolean equals(Object otherObject) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(firstName, lastName, group.getId());
+    return Objects.hash(firstName, lastName);
```

Comment:
hmm i am wrong combination of firstname & lastname & groupid isnt business key,  we can have 2 students with same firstname and lastname in the same group, such a coincidence is a possible ;D